modules/post/windows/gather/credentials/mdaemon_cred_collector.rb

```diff
@@ -0,0 +1,315 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+require 'rex'
+require 'base64'
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::Windows::Registry
+  Rank = ExcellentRanking
+  def initialize(info = {})
+    super(update_info(info,
+        'Name'          => 'Windows Gather MDaemonEmailServer Credential Cracking',
+        'Description'   => %q{
+          Finds and cracks the stored passwords of MDaemon Email Server.
+        },
+        'References'     =>
+        [
+          ['BID', '4686']
+        ],
+        'License'       => MSF_LICENSE,
+        'Author'        => ['Manuel Nader #AgoraSecurity'],
+        'Platform'      => ['win'],
+        'Arch'          => ['x64','x86'],
+        'SessionTypes'  => ['meterpreter', 'shell']
+    ))
+
+    register_options(
+      [OptString.new('RPATH', [false, 'Path of the MDaemon installation', false]) # If software is installed on a rare directory
+    ], self.class)
+  end
+
+  def run
+    if session.type != 'meterpreter'
+      print_error ('Only meterpreter sessions are supported by this post module')
+      return
+    end
```

Comment:
Remove the shell from SessionTypes on line 27 and you won't have to check for the session type manually. It will be automatically validated at the time of module run.